app/src/main/java/com/nervousfish/nervousfish/modules/pairing/AndroidAcceptThread.java

```diff
@@ -20,22 +17,23 @@
  * like a server-side Bluetooth client. It runs until a connection is accepted
  * (or until cancelled).
  */
-class AndroidAcceptThread extends Thread {
+final class AndroidAcceptThread extends Thread {
     private static final Logger LOGGER = LoggerFactory.getLogger("AndroidAcceptThread");
     // The local server socket
     private final BluetoothServerSocket serverSocket;
-    private final IServiceLocator serviceLocator;
+    private final AndroidBluetoothHandler bluetoothHandler;
 
     /**
      * Constructs a new AndroidAcceptThread that runs while listening for incoming connections. It behaves
      * like a server-side client. It runs until a connection is accepted
      * (or until cancelled).
      *
+     * @param bluetoothHandler The {@link AndroidBluetoothHandler} owning this thread
      * @param serviceLocator The service locator to be used
      */
-    AndroidAcceptThread(final IServiceLocator serviceLocator) {
+    AndroidAcceptThread(final AndroidBluetoothHandler bluetoothHandler, final IServiceLocator serviceLocator) {
```

Comment:
Because I didn't get the Eventbus working, as you saw yesterday. Now I'm using a callback, but I want that only the helper classes like AndroidAcceptThread can call these. Therefore I cannot implement them in the interface.